prover/src/proof.rs

```diff
@@ -0,0 +1,59 @@
+use super::{StackOutputs, StarkProof};
+use core::ops::Deref;
+
+// MIDEN PROOF
+// ================================================================================================
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MidenProof {
+    outputs: StackOutputs,
+    proof: StarkProof,
+}
```

Comment:
I'm not sure we need this struct - or at least not in this shape. I would not combine `outputs` and `proof` into a single object as they could come from different places.

Reply:
Done!

Reply:
Sounds good; the idea is to always avoid returning tuples as they can sometimes be confusing and cannot be documented. However, that isn't mandatory for this PR and was just an extra improvement